demes/src/specification.rs

```diff
@@ -1059,6 +1059,49 @@ impl Deme {
     pub fn proportions(&self) -> &[Proportion] {
         &self.proportions
     }
+
+    /// Size of Deme at a given time
+    pub fn size_at<F: Into<f64>>(&self, time: F) -> Result<DemeSize, DemesError> {
+        let time: f64 = time.into();
+
+        if time == f64::INFINITY && self.start_time == f64::INFINITY {
+            return Ok(self.epochs()[0].start_size);
+        }
+
+        let epochs: Vec<Epoch> = self
```

Comment:
I think you should be able to use [find](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find) here on the iterator.  It is late and I'm tired, but I think you can .map each item to send its time interval to find, ending up with Option<time interval> at the end.

Reply:
Example: 

```rust
fn main() {
    let v = vec![1, 2, 3, 4];

    let i = v.iter().map(|x| x*x).find(|x| x > &5);
    assert_eq!(i, Some(9));
}
```

Reply:
Used find